src/Sarif.UnitTests/Visitors/InsertOptionalDataVisitorTests.cs

```diff
@@ -110,101 +110,230 @@ public void InsertOptionalDataVisitor_PersistsAll()
                 OptionallyEmittedData.FlattenedMessages);
         }
 
-        [Fact]
-        public void InsertOptionalDataVisitorTests_FlattensGlobalMessageString()
-        {
-            string ruleId = nameof(ruleId);
-            string globalMessageId = nameof(globalMessageId);
-            string globalMessageValue = nameof(globalMessageValue);
+        private const int RuleIndex = 0;
+        private const string RuleId = nameof(RuleId);
+        private const string NotificationId = nameof(NotificationId);
+
+        private const string SharedMessageId = nameof(SharedMessageId);
+        private const string SharedKeyRuleMessageValue = nameof(SharedKeyRuleMessageValue);
+        private const string SharedKeyGlobalMessageValue = nameof(UniqueGlobalMessageValue);
+
+        private const string UniqueRuleMessageId = nameof(UniqueRuleMessageId);
+        private const string UniqueRuleMessageValue = nameof(UniqueRuleMessageValue);
+
+        private const string UniqueGlobalMessageId = nameof(UniqueGlobalMessageId);
+        private const string UniqueGlobalMessageValue = nameof(UniqueGlobalMessageValue);
 
+        private static Run CreateBasicRunForMessageStringLookupTesting()
+        {
+            // Returns a run object that defines unique string instances both
```

Comment:
>Returns [](start = 15, length = 7)

This is a good comment.

Ultra-nit: IMO it is a method summary comment so it belongs outside the method body. (I am _not_ suggested XML docs.)